lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -96,9 +93,12 @@ def rabbitmq_settings
 
         s[:timeout] = @connection_timeout || 0
         s[:heartbeat] = @heartbeat || 0
-        s[:tls] = @ssl if @ssl
-        s[:tls_certificate_path] = @tls_certificate_path || ""
-        s[:tls_certificate_password] = @tls_certificate_password || ""
+
+        # This MUST be passed as a string or the cert options will be ignored
+        s[:tls] = @ssl.is_a?(String) ? @ssl : "TLSv1.2"
```

Comment:
Can we default this in the config setting to "TLSv1.2" ?

```
config :ssl, :validate => :string, :default => "TLSv1.2"
```

Additionally, we could restrict validation to a set of known values, like:

```
config :ssl, :validate => [ "TLSv1.2", "TLSv1.1", "TLSv1" ], :default => "TLSv1.2"
```

Thoughts?